src/resolution/composite-arns-resolver.ts

```diff
@@ -80,20 +85,52 @@ export class CompositeArNSResolver implements NameResolver {
     }
 
     try {
-      const baseNameInCache =
-        await this.arnsNamesCache.getCachedArNSBaseName(baseName);
+      // start name list request before checking cache so that the name list
+      // stays fresh even when there are cache hits
+      const baseNameInCachePromise = this.arnsNamesCache
```

Comment:
nit: structure wise these may be useful as private or functions outside the class, not necessary

Reply:
Going to save this one for future refactoring.